tests/test_barostat_water_density.py

```diff
@@ -36,17 +35,13 @@
 
     # generate an alchemical system of a waterbox + alchemical ligand:
     # effectively discard ligands by running in AbsoluteFreeEnergy mode at lambda = 1.0
-    mol_a, _, core, ff = hif2a_ligand_pair.mol_a, hif2a_ligand_pair.mol_b, hif2a_ligand_pair.core, hif2a_ligand_pair.ff
-    complex_system, complex_coords, complex_box, complex_top = build_water_system(
-        initial_waterbox_width.value_in_unit(unit.nanometer)
-    )
-
-    min_complex_coords = minimize_host_4d([mol_a], complex_system, complex_coords, ff, complex_box)
-    afe = AbsoluteFreeEnergy(mol_a, ff)
-
-    unbound_potentials, sys_params, masses, coords = afe.prepare_host_edge(
-        ff.get_ordered_params(), complex_system, min_complex_coords
+    mol_a, ff = (
+        hif2a_ligand_pair.mol_a,
+        hif2a_ligand_pair.ff,
     )
+    unbound_potentials, sys_params, masses, coords, complex_box = get_solvent_phase_system(mol_a, ff, margin=0.0)
+    n_lig_atoms = get_romol_conf(mol_a).shape[0]
+    n_water_mols = (coords.shape[0] - n_lig_atoms) // 3
```

Comment:
This seems non-obvious and is repeated in a few places. Would it make sense to introduce a utility or return this directly from `get_solvent_phase_system` (feel free to ignore if not relevant, I'm not super familiar with this part of the code)

Reply:
Updated to simplify (this is equivalent to: `n_water_mols = len(group_indices) - 1`).